### PR TITLE
Refactoring Config Form

### LINKED ISSUE #1907: UX: 非法服务器地址
### Shadowsocks version / 影梭版本
Windows 4.0.10

### Environment(Operating system, .NET Framework, etc) / 使用环境（操作系统，.NET Framework等）
Windows10 对应

### Steps you have tried / 操作步骤
我的SS服务器给的地址为 aa.bb.cc.dd 的形式，会报非法服务器地址。
用IP写入后，点击添加按钮，可以在左侧列表中显示。但是点击上一个服务器地址，同样会报服务器地址非法。

### PR DESCRIPTION
**Pull request by Stzx**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

- [x] [Searched](https://github.com/shadowsocks/shadowsocks-windows/search?q=is%3Apr&type=Issues) for similar pull requests
- [x] Compiled the code with Visual Studio

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New feature

---

### Description of your *pull request* and other information

Fix #1907 

Still need further testing.

😀

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

### CODE REVIEW

**Comment by Stzx** [comment]
Consider upgrading the version of the appveyor mirror ? 

The current `.Net Framework` version of the project is `4.6.2`, and compiling with the VS 2015 image version is obviously not suitable for the current `.Net Framework 4.6.2` version.

If I don't upgrade, I roll back and modify the code.

**Comment by celeron533** [comment]
Sorry, I'm too busy these days. I'll check it soon.

File: shadowsocks-csharp/View/ConfigForm.cs [hunk 1]
```diff
@@ -88,23 +89,31 @@ private void ConfigValueChanged(object sender, EventArgs e)
             ApplyButton.Enabled = true;
         }
 
-        private bool ValidateAndSaveSelectedServerDetails()
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```
Comment (celeron533) [inline shadowsocks-csharp/View/ConfigForm.cs]: Force compiler to inline for performance boost?

Comment (Stzx) [reply shadowsocks-csharp/View/ConfigForm.cs]: 这里目前看来，不会得到性能上的提升，可以在合并时将其剔除。

Comment (celeron533) [reply shadowsocks-csharp/View/ConfigForm.cs]: Never mind. Could leave it.

**Pull request merged**